### DISCUSSION TITLE
Routing requests to existing express application in hooks.js

### DISCUSSION THREAD

**Discussion by trenta3** [sveltejs/kit#4838] [Q&A] [answered]
I'm trying to use an existing express application inside SvelteKit response handler. I detail here a minimal example.

```bash
cd /tmp
npm init svelte@next test
# Chose Skeleton Project
cd test && npm install
npm install -D express
```
Then we fake an existing express application in `src/lib/server.js`:
```javascript
import express from 'express'

export let app = express()
app.get("/status", (req, res) => res.send("Status OK!"))
```

How to set things up in `src/hooks.js` to try to route requests through the express app, and otherwise send them to the usual sveltekit framework?
```javascript
import { app } from '$lib/server.js'

export async function handle({ event, resolve }) {
    return app(event.request, resolve, async () => {
	return await resolve(event)
    })
}
```
I thought this should be a way to let the express application answer, and if the application can't then we pass as next method the usual SvelteKit resolve.
The result is that i receive a "Not found" both for "/" and for "/status".
Can someone point me in the right direction?

**Comment by benmccann** [comment, marked as answer]
`handle` needs to return a `Response`, but it looks like you have it returning an Express app?

If you're using `adapter-node` you could use a custom server and add any middleware there: https://kit.svelte.dev/faq#integrations-how-do-i-use-middleware

**Reply by trenta3** [reply]
Thank you! This works seamlessly!
